### DISCUSSION TITLE
Is it possible to probe/measure the energy used on the Loihi 1 with LAVA?

### DISCUSSION THREAD

**Discussion by JonathanGundorph** [lava-nc/lava#432] [Q&A] [answered]
On the Loihi 1 chip, there are some physical probes that measures the consumed energy, i.e. the current that the chip absorbs when you do the inference, when running a SNN/SCNN. I would like to have access to those measurements, so I can calculate the amount of energy the Loihi consumes during inferences, which require some software infrastructure to do so. 

So far, I know it's possible to access on the Loihi API. But my question is, is LAVA capable of doing this energy estimation as well, and if not, is this on the roadmap to be implemented at some point?

Kind Regards
Jonathan

**Comment by PhilippPlank** [comment, marked as answer]
Hello Jonathan,

thanks for the question. We are currently working on the profiling tools and API for Lava, which allow execution time and energy measurements for Loihi 2 systems. This feature should be available on the next release, together with a tutorial on how to use it - the plan is to have it released before the INRC workshop on 2nd of November.

Keep in mind, Lava does not yet support Loihi 1.

I hope this answers your question :)

**Reply by JonathanGundorph** [reply]
Hello Philipp,

Thank you very much for your answer. As an additional question, is it on your roadmap to support Loihi 1 sometime in the future?

Kind Regards
Jonathan

**Reply by PhilippPlank** [reply]
Yes, Loihi 1 support for Lava and afterwards also supporting the profiling tools for Loihi 1 is planned, but I cannot give you an estimate when it will be ready.

**Reply by awintel** [reply]
@JonathanGundorph We expect to start working on Loihi 1 support in December. Thus, the current expectation is that it should be available sometime in January.
